src/Microsoft.AspNetCore.Mvc/DesignTimeMvcServiceCollectionProvider.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.FileProviders;
+
+namespace Microsoft.AspNetCore.Mvc
+{
+    public class DesignTimeMvcServiceCollectionProvider
+    {
+        public static IServiceCollection PopulateServiceCollection(IServiceCollection services, string assemblyName)
+        {
+            var hostingEnvironment = new HostingEnvironment
+            {
+                ApplicationName = assemblyName
```

Comment:
This is super hacky isn't it? The `assemblyName` here could totally be `Microsoft.AspNetCore.Mvc.TagHelpers`. That would mean we'd be telling MVC that the users hosted application is a class library O_o.

Is there another route to get ApplicationParts to recognize a specific assembly?